addons/uom/static/src/components/many2x_uom_tags/many2x_uom_tags.js

```diff
@@ -13,6 +13,12 @@ import { UomAutoComplete } from "@uom/components/uom_autocomplete/uom_autocomple
 import { roundPrecision } from "@web/core/utils/numbers";
 import { onWillUpdateProps } from "@odoo/owl";
 
+function _getProductRelatedModel() {
+    const field = this.env.model.config.fields[this.props.productField];
+    let resModel = field?.relation || this.props.record.resModel;
+    return ["product.product", "product.template"].includes(resModel) ? resModel : "product.product";
+}
```

Comment:
Overall looks good to me. However, I'm afraid of some edge case that may happen later (it doesn't exist now), which is that if we have a child model (like an order line) that's related to `product.template` not `product.product`. In that case the value of `field` will be `undefined`, because `this.env.model.config.fields` always returns information about the root model fields not the current model. Therefore `resModel` will fallback to the default value `product.product` which will trigger a searchRead to a `product.product` with the id of a `product.template`.

Currently the only model that uses the widget and is related to `product.template` is `mrp.bom` and it works fine because it's the root model, so the `field` value gets set correctly.